cardano-c/src/wallet.rs

```diff
@@ -25,27 +25,30 @@ use types::{AccountPtr, WalletPtr};
 /// object. This function may fail if:
 ///
 /// - panic: if there is no more memory to allocate the object to return
-/// - panic or return 0 (nullptr or NULL) if the given seed_ptr is of invalid length
+/// - return `failure` if the given seed_ptr is of invalid length
+/// - return 'success' when not
 ///
 #[no_mangle]
 pub extern "C" fn cardano_wallet_new(
     entropy_ptr: *const u8,  /* expecting entropy ptr ... */
     entropy_size: usize,     /* entropy size */
     password_ptr: *const u8, /* password ptr */
     password_size: usize,    /* password size */
-) -> WalletPtr {
+    wallet_out: *mut WalletPtr,
+) -> CardanoResult {
     let entropy_slice = unsafe { slice::from_raw_parts(entropy_ptr, entropy_size) };
     let password = unsafe { slice::from_raw_parts(password_ptr, password_size) };
 
     let entropy = match bip::bip39::Entropy::from_slice(entropy_slice) {
-        Err(_) => return ptr::null_mut(),
+        Err(_) => return CardanoResult::failure(),
         Ok(e) => e,
     };
 
     let wallet = bip44::Wallet::from_entropy(&entropy, &password, hdwallet::DerivationScheme::V2);
 
     let wallet_box = Box::new(wallet);
-    Box::into_raw(wallet_box)
+    unsafe { ptr::write(wallet_out, Box::into_raw(wallet_box)) };
```

Comment:
might be a good idea to check the wallet_out pointer being non NULL and returning a friendly CardanoResult error if that's the case ?

Reply:
I see... The behavior is undefined in that case, wouldn't a panic with a message be better though? I think a null pointer here is most probably a bug in the caller?
Should I always check for null pointers?